Bogus vscode error

Razz4780 opened the pull request:
Fixes #1504 

An error notification was shown every time the extension was run (on every `close` from the child process stderr). Added a check for stderr being empty.

Also improved error notifications a bit.

aviramha commented:
The stderr can have warnings or logs if logging enabled, so not sure if this condition is smart.

Razz4780 commented:
> The stderr can have warnings or logs if logging enabled, so not sure if this condition is smart.

Hmm, that's right. We use the same check for `mirrord ls`, so I assumed it would be correct. Looks like we have to ignore stderr completely if it doesn't match the error pattern?

aviramha commented:
> > The stderr can have warnings or logs if logging enabled, so not sure if this condition is smart.
> 
> Hmm, that's right. We use the same check for `mirrord ls`, so I assumed it would be correct. Looks like we have to ignore stderr completely if it doesn't match the error pattern?

I guess - I thought maybe checking error code but even if mirrord was okay the running process might return error exit code.

Razz4780 commented:
Made some improvements to the extension code:
1. Found a bug where an error notification was not shown if two notifications were fired quickly. Fixed it by including error message in the general failed/get help notification.
2. Removed use of `exec` from the code. `exec` takes the command as a string, so we had to escape whitespace
3. Added inspecting error codes and signals

infiniteregrets reviewed (approved):
looks good overall! small nit

infiniteregrets commented:
The e2e is not running, the fix is in #1528, ill move it to a separate pr
